nixos/modules/services/networking/anubis.nix

```
@@ -224,6 +238,9 @@ in
         and socket paths.
       '';
       type = types.attrsOf (types.submodule (commonSubmodule false));
+
+      # Merge defaultOptions into each instance
+      apply = (i: lib.mapAttrs (_: instance: lib.recursiveUpdate cfg.defaultOptions instance) i);
```

eta-reduction here
```suggestion
      apply = lib.mapAttrs (_: lib.recursiveUpdate cfg.defaultOptions);
```

Also correct me if I'm wrong but does this not make the custom `mkDefaultOption` redundant?

---

I mean, it does partially make `mkDefaultOption` reduntant, but it's unfortunately it is still needed the way the module is currently set up, since each instance has it's own defaults for the defined settings and those will overwrite the ones set in `cfg.defaultOptions`. I couldn't think of a better way to do this without completely re-doing the options definitions, which I didn't want to do for a quick fix, but if you have any ideas for how to do this better I'd be more than happy to give them a shot.

---

I'll take a look into this later. :D